taptargetview/src/main/java/com/getkeepsafe/taptargetview/TapTargetView.java

```diff
@@ -487,6 +536,46 @@ public void onClick(View v) {
             (int) lastTouchX, (int) lastTouchY);
         final boolean clickedInsideOfOuterCircle = distanceToOuterCircleCenter <= outerCircleRadius;
 
+        if (target.showSequencePagination && clickedInsideOfOuterCircle) {
+          if (target.sequenceCurrentTargetIndex < target.sequenceTargetCount) {
+
+            int buttonHeight;
+            if (skipLayout != null && nextLayout != null) {
+              buttonHeight = Math.max(skipLayout.getHeight(), nextLayout.getHeight());
+            } else if (skipLayout != null) {
+              buttonHeight = skipLayout.getHeight();
+            } else {
+              buttonHeight = nextLayout.getHeight();
+            }
+
+            boolean clickedInSkipText = skipLayout != null &&
+                (lastTouchX > (skipTextXPosition - BUTTON_PADDING) && lastTouchX < (nextTextPosition + BUTTON_PADDING))
+                && (lastTouchY > (textBounds.bottom - buttonHeight - BUTTON_PADDING) && lastTouchY < (textBounds.bottom + BUTTON_PADDING));
+
+
+            boolean clickedInNextText = nextLayout != null &&
+                (lastTouchX > (nextTextPosition - BUTTON_PADDING) && lastTouchX < (textBounds.right + BUTTON_PADDING))
+                && (lastTouchY > (textBounds.bottom - buttonHeight - BUTTON_PADDING) && lastTouchY < (textBounds.bottom + BUTTON_PADDING));
+
+            if (clickedInNextText) {
+              isInteractable = false;
+              listener.onTargetClick(TapTargetView.this);
+            } else if (clickedInSkipText && cancelable) {
+              isInteractable = false;
+              listener.onTargetCancel(TapTargetView.this);
+            }
+          } else {
+            final boolean clickedInDoneText = doneLayout != null &&
```

Comment:
Maybe  you should make `clickedInSkipText` and `clickedInNextText` also `final`

Reply:
Agree, it's better to keep the code pattern